### PR TITLE
DF-20626 add bitgo-reserves EA

### PR DESCRIPTION
**Pull request by mmcallister-cll**
## Description
Adding new EA for bitgo reserves
**NOTE** the default endpoint is a test endpoint and not meant to be used in production
**NOTE** no secrets

## Steps to Test
1. yarn test bitgo-reserves
2. sanity test reserves endpoint EA request

## Quality Assurance

- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `adapter-secrets` configuration file or update the [soak testing blacklist](/packages/scripts/src/get-changed-adapters/soakTestBlacklist.ts).
- [ ] If a new adapter was made, or a new endpoint was added, update the `test-payload.json` file with relevant requests.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Jira.
- [ ] This is related to a maximum of one Jira story or GitHub issue.
- [ ] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types).
- [ ] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

[DF-20626]: https://smartcontract-it.atlassian.net/browse/DF-20626?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/sources/bitgo-reserves/src/config/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { AdapterConfig } from '@chainlink/external-adapter-framework/config'
+
+export const config = new AdapterConfig({
+  API_ENDPOINT: {
+    description: 'An API endpoint for Data Provider',
+    type: 'string',
+    default: 'https://d2nz84bmzl209r.cloudfront.net',
```
Comment (mxiao-cll) [inline packages/sources/bitgo-reserves/src/config/index.ts:7]: I think we should not set the testing endpoint as default?

File: packages/sources/bitgo-reserves/src/endpoint/reserves.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
```
Comment (mxiao-cll) [inline packages/sources/bitgo-reserves/src/endpoint/reserves.ts:1]: We can use PoRProviderResponse and PoRProviderEndpoint here

File: packages/sources/bitgo-reserves/src/transport/reserves.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          response: {
+            errorMessage: `The data provider didn't return any value`,
+            statusCode: 502,
+          },
+        }
+      })
+    }
+
+    return params.map((param) => {
+      const result = response.data.totalReserve
```
Comment (mxiao-cll) [inline packages/sources/bitgo-reserves/src/transport/reserves.ts:45]: We can read time here and send as part of PoRProviderResponse

File: packages/sources/bitgo-reserves/src/config/index.ts (hunk 1 above)
Comment (mmcallister-cll) [reply packages/sources/bitgo-reserves/src/config/index.ts:7]: so set the real one and overwrite with the test one? or just hold until the real one is ready?

File: packages/sources/bitgo-reserves/src/transport/reserves.ts (hunk 3 above)
Comment (mmcallister-cll) [reply packages/sources/bitgo-reserves/src/transport/reserves.ts:45]: good spot

File: packages/sources/bitgo-reserves/src/endpoint/reserves.ts (hunk 2 above)
Comment (mmcallister-cll) [reply packages/sources/bitgo-reserves/src/endpoint/reserves.ts:1]: Debatable if there's any value. The advantage of PoRProviderResponse is to force returning a `ripcord` field, which this API+endpoint does/will not have (as far as we know at this point).

Given that we'd just end up forcing ripcord=false, what do you think?

Comment (mxiao-cll) [reply packages/sources/bitgo-reserves/src/endpoint/reserves.ts:1]: I am fine with both

File: packages/sources/bitgo-reserves/src/config/index.ts (hunk 1 above)
Comment (mmcallister-cll) [reply packages/sources/bitgo-reserves/src/config/index.ts:7]: rather set the real one than blank

File: packages/sources/bitgo-reserves/src/endpoint/reserves.ts (hunk 2 above)
Comment (mmcallister-cll) [reply packages/sources/bitgo-reserves/src/endpoint/reserves.ts:1]: alright, i'm leaning towards without PoRProviderResponse. Happy to revisit if the API changes and they decide to add a ripcord in the future.

**Review by mxiao-cll — APPROVED** [review]

**Pull request merged**